workers/unity/Packages/io.improbable.gdk.buildsystem/Configuration/BuildConfigEditor.cs

```diff
@@ -633,24 +633,19 @@ private GUIContent GetBuildTargetGuiContents(BuildTargetConfig c)
 
         private void DrawBuildTargets(BuildEnvironmentConfig env, int hash)
         {
-            // Build a mapping the build target position in the toolbar and it's position in the BuildTargets array.
-            // Sometimes there is a different of targets in the BuildTargets array and those in the toolbar as
-            // some are deprecated
+            // Build a mapping from the 'build target' position in the toolbar to it's position in the 'BuildTargets' array.
+            // We do this since sometimes there are different number of targets in the BuildTargets array and those
+            // in the toolbar as some targets are deprecated
             if (toolbarMapping.Count == 0)
             {
-                var index = 0;
-                var skipped = 0;
-                env.BuildTargets.ForEach(t =>
+                var toolbarIndex = 0;
+                for (var buildTargetIndex = 0; buildTargetIndex < env.BuildTargets.Count; buildTargetIndex++)
                 {
-                    if (t.Deprecated)
+                    if (!env.BuildTargets[buildTargetIndex].Deprecated)
                     {
-                        skipped++;
+                        toolbarMapping.Add(toolbarIndex++, buildTargetIndex++);
```

Comment:
I don't think you should be incrementing `buildTargetIndex` here?

Reply:
Oops! Very true, I'll get that fixed